LongModuleFinder/main.cpp

```diff
@@ -0,0 +1,129 @@
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <CppBuildInsights.hpp>
+
+using namespace Microsoft::Cpp::BuildInsights;
+using namespace Activities;
+using namespace SimpleEvents;
+
+class LongModuleFinder : public IAnalyzer
+{
+    struct IdentifiedCompilerInvocation
+    {
+        unsigned InvocationId;
+        double InvocationTime;
+
+        bool operator<(const IdentifiedCompilerInvocation& other) const {
+            return InvocationTime > other.InvocationTime;
+        }
+    };
+
+public:
+    LongModuleFinder() :
+        pass_{ 0 },
+        cachedInvocationDurations_{},
+        identifiedCompilerInvocations_{}
+    {}
+
+    AnalysisControl OnBeginAnalysisPass() override
+    {
+        ++pass_;
+        return AnalysisControl::CONTINUE;
+    }
+
+    AnalysisControl OnStopActivity(const EventStack& eventStack)
+        override
+    {
+        if (pass_ == 1) {

```

Comment:
It is possible to get the top modules, header units and PCHs in a single pass instead of 2 by:

1. Caching the `EventInstanceId` of the `FrontEndPass` during the Module, HeaderUnit, or PrecompiledHeader simple event.
2. At the stop event for the `FrontEndPass` activity, if the `EventInstanceId` of the current `FrontEndPass` has been saved in step 1, take note of the duration of the `FrontEndPass` and the file name.

Reply:
Fixed.